src/sumo_store_riak.erl

```diff
@@ -705,3 +707,14 @@ build_sort(Sorts) ->
     binary:list_to_bin([sumo_utils:to_bin(Field), "_register", " ", sumo_utils:to_bin(Dir)])
   end || {Field, Dir} <- Sorts],
   [{sort, binary:list_to_bin(interpose(", ", Res))}].
+
+-spec count_by(DocName, Conditions, State) -> Response when
+  DocName    :: sumo:schema_name(),
+  Conditions :: sumo:conditions(),
+  State      :: state(),
+  Response   :: sumo_store:result(non_neg_integer(), state()).
+count_by(DocName, [], State) ->
+  count(DocName, State);
+count_by(_DocName, _Conditions, #{conn := _Conn} = _State) ->
+    0.
```

Comment:
This should only have 2 spaces for indentation.

Reply:
Let's move this function to the `API` section; below `count`.

On the other hand, I don't really see the point of having this function, at least I don't see specific logic with the conditions, for example, if condition list if empty, bypass to `count(DocName, State)`, otherwise return `0`. This looks to me like a very specific logic. Could you please elaborate more on this, what is the purpose of having this function implemented in this way? What if we have conditions, should we perform the count query based on the conditions?

Reply:
currrrently I'm just getting the code to  compile and run, I'm not sure how to implement a count_by in the context, maybe I could do a search and get the count ?

Reply:
> currrrently I'm just getting the code to compile and run, I'm not sure how to implement a count_by in the context, maybe I could do a search and get the count ?

Sorry, I cannot help much here since I haven't worked with Riak a long time ago, but I'd review RiakKV doc and see what would be the best way for implementing this, maybe investigate if there are examples or other cases where this has been implemented already.

Once you have some implementation done, we can review it! Looking forward to that 😉